src/ORM/Connect/NullDatabase.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+namespace SilverStripe\ORM\Connect;
+
+use BadMethodCallException;
+use Exception;
+
+/**
+ * Utility class required due to bad coupling in framework.
```

Comment:
Is it due to bad coupling? I think this is a [perfectly reasonable thing](https://en.wikipedia.org/wiki/Null_object_pattern#Description)? 

As above, would it be possible to create a DatabaseInterface and implement that rather than extending Database?

Reply:
It's more about the composition of the kernel. The kernel is designed to be composed with all these different layers so that in theory you could design your own lightweight application, but it can't function without a database, which seems like unnecessary coupling. Not sure it's "bad" because decoupling it probably would have been a massive amount of work. 

I think "naive implementation" is a better descriptor here than the null object pattern. Maybe naive database would make more sense.

Reply:
NullDatabase makes much more sense IMO, I understand it as a class that do not connect to a database and exists only to satisfy API signatures

NaiveDatabase I would think of more as a quickly hacked up database connection class

Reply:
Well a lot of the methods are stubs that just return empty strings, etc. But the most critical methods (e.g. query) throw, so i dunno. NullaiveDatabase? 🤣

Reply:
Throwing exceptions makes it much more non-functional/null than naive which is just inefficient

Reply:
Not sure there's anything actionable in this conversation.